src/components/TableCreation.tsx

```diff
@@ -0,0 +1,163 @@
+};
+
+const TableCreation = ({ rows, rowContent, columns, tableId, customTableWidth }: any) => {
+  const { currentTheme } = useCurrentTheme();
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(5);
+  const [orderBy, setOrderBy] = React.useState<any>();
+
+  const totalRows = rows.length;
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  const [sortDirection, setSortDirection] = React.useState<'asc' | 'desc'>('asc');
+  const [sortedColumn, setSortedColumn] = React.useState<any>('id');
+
+  const fixedHeaderContent = () => {
+    return (
+      <TableRow>
+        {columns.map((column: any) => (
+          <TableCell
+            key={column.dataKey}
+            variant="head"
+            align={column.dataKey == 'address' ? 'left' : 'right'}
+            style={{ width: column.width }}
+            sx={{
+              backgroundColor: colors[currentTheme].primary,
+              color: colors[currentTheme].secondary,
+            }}
+          >
+            <TableSortLabel
+              sx={{
+                ':hover': {
+                  color: colors[currentTheme].secondary,
+                },
+                ':focus': {
+                  color: colors[currentTheme].secondary,
+                },
+                ':active': {
+                  color: colors[currentTheme].secondary,
+                },
+              }}
+              active={orderBy === column.dataKey}
+              direction={orderBy === column.dataKey ? sortDirection : 'asc'}
+              onClick={() => handleSortClick(column.dataKey)}
+            >
+              {column.label}
+            </TableSortLabel>
+          </TableCell>
+        ))}
+      </TableRow>
+    );
+  };
+
+  const handleSortClick = (column: any) => {
+    if (column === sortedColumn) {
+      setSortDirection(sortDirection === 'asc' ? 'desc' : 'asc');
+    } else {
+      setSortedColumn(column);
+      setSortDirection('asc');
+      setOrderBy(column);
+    }
+  };
+
+  const sortedRows = React.useMemo(() => {
+    const sortedRows = [...rows];
+    sortedRows.sort((rowA, rowB) => {
+      const valueA = rowA[sortedColumn];
+      const valueB = rowB[sortedColumn];
+      if (sortDirection === 'asc') {
+        return valueA < valueB ? -1 : valueA > valueB ? 1 : 0;
+      } else {
+        return valueB < valueA ? -1 : valueB > valueA ? 1 : 0;
+      }
+    });
+    return sortedRows;
+  }, [rows, sortedColumn, sortDirection]);
+
+  const screenWidth = (window.innerWidth * parseFloat(customTableWidth)) / 100;
+  const tableWidth = 1241.6 * (parseFloat(customTableWidth) / 100);
+  const rowHeight = tableId === 'waiting' || tableId === 'miners' ? 64.8 : 52.813;
+  const headerHeight = 57.9;
+  const tableHeight =
```

Comment:
Removed tableWidth, it was because horizontal scroll added height to the table, and made it scrollable vertically. However, they only applied to my screen size, so other than that it's useless. However, I am keeping the rowHeight in order to make the table scrollable on page instead of keeping it in a fixed box (on deployer's instructions) (there might be a way to get the row height dynamically, but I don't know it yet)